web/modules/custom/joinup_core/joinup_core.install

```diff
@@ -60,3 +60,41 @@ function joinup_core_requirements($phase) {
 function joinup_core_update_8100(&$sandbox) {
   \Drupal::service('module_installer')->install(['config_sync']);
 }
+
+/**
+ * Use the 'private' scheme for contact form uploads.
+ */
+function joinup_core_update_8101() {
+  include_once DRUPAL_ROOT . '/core/includes/install.inc';
+  $settings_file = DRUPAL_ROOT . '/sites/default/settings.php';
+
+  // The private files directory should have been already created.
+  if (!$private_files_directory = realpath(DRUPAL_ROOT . '/../private')) {
+    throw new \RuntimeException("Directory '$private_files_directory' has to be created prior running this script.");
+  }
+
+  // As the private files directory is outside the web tree, we really don't
+  // need a .htaccess file but we still provide it to comply with the Drupal
+  // security policy in the unlikely case this directory will be moved under the
+  // web tree in the future.
+  file_save_htaccess($private_files_directory);
+
+  // Save current settings in a local variable.
+  $settings = Settings::getAll();
+
+  // Add the 'file_private_path' in settings.php.
+  $private_files_directory_settings['settings']['file_private_path'] = (object) [
+    'value' => $private_files_directory,
+    'required' => TRUE,
+  ];
+  chmod($settings_file, 0666);
+  drupal_rewrite_settings($private_files_directory_settings, $settings_file);
```

Comment:
If a website is correctly configured then this should be impossible. The site should not be allowed to overwrite its own settings file, since this would make it easier to do privilege escalation attacks.